`protoc-gen-openapi` crashes if method in service does not contain `google.api.http` option

I have a protobuf / gRPC service containing both methods with the `google.api.http` and without. This seems to crash `protoc-gen-openapi`. If the service only contains annotated methods, it works fine.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x48 pc=0x1030faf80]

goroutine 1 [running]:
github.com/googleapis/gnostic/apps/protoc-gen-openapi/generator.(*OpenAPIv3Generator).addPathsToDocumentV3(0x140002c0b40, 0x14000177380, 0x140002cb600)
	/Users/oxisto/go/pkg/mod/github.com/googleapis/gnostic@v0.5.6-0.20210515010530-c686e505ec47/apps/protoc-gen-openapi/generator/openapi-v3.go:136 +0x170
github.com/googleapis/gnostic/apps/protoc-gen-openapi/generator.(*OpenAPIv3Generator).buildDocumentV3(0x140002c0b40, 0x10324f2e0)
	/Users/oxisto/go/pkg/mod/github.com/googleapis/gnostic@v0.5.6-0.20210515010530-c686e505ec47/apps/protoc-gen-openapi/generator/openapi-v3.go:83 +0x148
github.com/googleapis/gnostic/apps/protoc-gen-openapi/generator.(*OpenAPIv3Generator).Run(0x140002c0b40, 0x140002c0b40, 0x0)
	/Users/oxisto/go/pkg/mod/github.com/googleapis/gnostic@v0.5.6-0.20210515010530-c686e505ec47/apps/protoc-gen-openapi/generator/openapi-v3.go:57 +0x28
main.main.func1(0x1400028f720, 0x0, 0x140001ac0e0)
	/Users/oxisto/go/pkg/mod/github.com/googleapis/gnostic@v0.5.6-0.20210515010530-c686e505ec47/apps/protoc-gen-openapi/main.go:25 +0x34
google.golang.org/protobuf/compiler/protogen.run(0x0, 0x0, 0x10328e1b0, 0x14000000180, 0x14000000180)
	/Users/oxisto/go/pkg/mod/google.golang.org/protobuf@v1.26.0/compiler/protogen/protogen.go:74 +0x194
google.golang.org/protobuf/compiler/protogen.Options.Run(0x0, 0x0, 0x10328e1b0)
	/Users/oxisto/go/pkg/mod/google.golang.org/protobuf@v1.26.0/compiler/protogen/protogen.go:52 +0x38
main.main()
	/Users/oxisto/go/pkg/mod/github.com/googleapis/gnostic@v0.5.6-0.20210515010530-c686e505ec47/apps/protoc-gen-openapi/main.go:24 +0x30
--openapi_out: protoc-gen-openapi: Plugin failed with status code 2.
```

To demonstrate the issue in a minimum viable crash, I have extended the example (https://github.com/google/gnostic/blob/master/apps/protoc-gen-openapi/examples/tests/bodymapping/message.proto) with a simple test method:
```diff
diff --git a/apps/protoc-gen-openapi/examples/tests/bodymapping/message.proto b/apps/protoc-gen-openapi/examples/tests/bodymapping/message.proto
index f8c350f..625252f 100644
--- a/apps/protoc-gen-openapi/examples/tests/bodymapping/message.proto
+++ b/apps/protoc-gen-openapi/examples/tests/bodymapping/message.proto
@@ -28,6 +28,8 @@ service Messaging {
             body: "text"
         };
     }
+
+    rpc Test(Message) returns (Message);
 }
 message Message {
     string message_id = 1;
```

I was using a simple `protoc --openapi_out=. tests/bodymapping/message.proto` from the `examples` folder. I am using `libprotoc 3.15.8`.

Any ideas?

--- a/apps/protoc-gen-openapi/generator/openapi-v3.go
+++ b/apps/protoc-gen-openapi/generator/openapi-v3.go
@@ -127,11 +127,12 @@ func (g *OpenAPIv3Generator) addPathsToDocumentV3(d *v3.Document, file *protogen
 			inputMessage := method.Input
 			outputMessage := method.Output
 			operationID := service.GoName + "_" + method.GoName
-			extension := proto.GetExtension(method.Desc.Options(), annotations.E_Http)
+			xt := annotations.E_Http
+			extension := proto.GetExtension(method.Desc.Options(), xt)
 			var path string
 			var methodName string
 			var body string
-			if extension != nil {
+			if extension != nil && extension != xt.InterfaceOf(xt.Zero()) {
 				rule := extension.(*annotations.HttpRule)
 				body = rule.Body
 				switch pattern := rule.Pattern.(type) {

